### ISSUE TITLE
numpydoc format docstrings

### ISSUE THREAD

**Issue by bmcfee** [craffel/mir_eval#104]
**tldr: changing the docstring format will make integration easier.**

## Background
I've implemented a [first cut](https://github.com/marl/jams/commit/4413d2dfe3192d875e04d16dbf1fb2b2ce0686b1) at JAMS<->mir_eval integration, as described in [this comment thread](https://github.com/marl/jams/issues/16) (paging @justinsalamon).

The basic idea here is that jams provides wrappers for the top-level mir_eval `evaluate()` functions, which do some basic sanity checking and schema validation on the input (JAMS) objects, and then convert them into mir_eval-friendly inputs.  This is implemented through a handful of decorator functions which get applied to each submodule's `evaluate`.  This is almost totally seamless, except for the docstrings.

## why should I have to change the docstring format, you jerk?

A few reasons this would be a good idea:

1. [numpydoc](https://github.com/numpy/numpydoc) format is more human-readable than raw sphinx
2. the compiled sphinx output looks the same (or better)
3. fits in with the numpy/scipy ecosystem
4. it would let JAMS's eval decorators use numpydoc's parsing functions to rip apart the docstring, modify the `Parameters` section, and put it back together.  Any changes to mir_eval's comments will propagate upstream automatically, and everything looks nice to the user.

**Comment by craffel** [comment]
Go for it!

**Comment by bmcfee** [comment]
... damn.

lil help?

**Comment by craffel** [comment]
https://github.com/dadadel/pyment ?

**Comment by bmcfee** [comment]
Nope, pyment doesn't correctly parse the current docstrings.

**Comment by bmcfee** [comment]
074eeb3f423d10ff9591aa395c59b5c40908607e

**Comment by craffel** [comment]
:+1:
